MapboxNavigation/NextBannerView.swift

```diff
@@ -91,30 +91,19 @@ open class NextBannerView: UIView {
         bottomSeparatorView.heightAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale).isActive = true
     }
     
-    func shouldShowNextBanner(for routeProgress: RouteProgress) -> Bool {
-        guard let upcomingStep = routeProgress.currentLegProgress.upComingStep else { return false }
-        
-        let durationForNext = RouteControllerHighAlertInterval * RouteControllerLinkedInstructionBufferMultiplier
-
-        guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining <= durationForNext, upcomingStep.expectedTravelTime <= durationForNext else {
-            return false
-        }
-        
-        guard let _ = upcomingStep.instructionsDisplayedAlongStep?.last else { return false }
-        
-        return true
-    }
-    
-    public func update(for routeProgress: RouteProgress) {
-        guard shouldShowNextBanner(for: routeProgress) else {
-            hide()
-            return
+    public func update(for currentStepProgress: RouteStepProgress) {
+        let distanceTraveled = currentStepProgress.distanceTraveled
+        guard let visualInstruction = currentStepProgress.currentVisualInstruction,
+              let tertiaryInstruction = visualInstruction.tertiaryInstruction,
+                  distanceTraveled <= visualInstruction.distanceAlongStep,
```

Comment:
This should not be necessary, this check should already be happening via in RouteController.swift and we should be using the currentVisualInstructionIndex, similar to how the main instruction is getting updated: 

https://github.com/mapbox/mapbox-navigation-ios/blob/f08eb8ee299dd8b8e71bf0c72adaa64c6c4dd174/MapboxNavigation/RouteMapViewController.swift#L345

Reply:
@bsudekum 😸  was already making those changes post the recent merge.